File: java/src/Structural/Bridge/SuperSizeSoda.java
package Structural.Bridge;


//SuperSizeSoda.java - two of two classes extending the Abstract

public class SuperSizeSoda extends Soda {  
   public SuperSizeSoda() {
       setSodaImp();
   }
   
   public void pourSoda() {
       SodaImp sodaImp = this.getSodaImp();
       for (int i = 0; i < 5; i++) {
           System.out.print("...glug...");
           sodaImp.pourSodaImp();
       }
       System.out.println(" ");       
   }
}

File: csharp/src/Structural/Bridge/SuperSizeSoda.cs
//SuperSizeSoda.java - two of two classes extending the Abstract


namespace Structural.Bridge {



using System;
using System.Collections.Generic;


public class SuperSizeSoda : Soda {  
   public SuperSizeSoda() {
       setSodaImp();
   }
   
   override public string pourSoda() {
       SodaImp sodaImp = this.getSodaImp();
		List<string> retval = new List<string>();
       for (int i = 0; i < 5; i++) {
           retval.Add("...glug...");
				retval.Add( sodaImp.pourSodaImp() );
       }
		return string.Join(" ",retval.ToArray());
   }
}

}
